burn-core/src/nn/conv/conv2d.rs

```diff
@@ -63,21 +68,32 @@ impl Conv2dConfig {
     pub fn init<B: Backend>(&self) -> Conv2d<B> {
         let shape = [
             self.channels[1],
-            self.channels[0],
+            self.channels[0] / self.groups,
```

Comment:
Does it mean we had a bug that the shape was not calculated correctly for groups > 1?

BTW, in PyTorch's implementation, the function panics if the first channel is not divisible, which we should do as well. Otherwise, the value floored, e.g. 11/4.